src/components/package/edit-custom/custom-package-edit.js

```diff
@@ -43,14 +43,24 @@ class CustomPackageEdit extends Component {
     showSelectionModal: false,
     allowFormToSubmit: false,
     packageSelected: this.props.initialValues.isSelected,
+    packageVisible: this.props.initialValues.isVisible,
     formValues: {}
   }
 
   componentWillReceiveProps(nextProps) {
     let wasPending = this.props.model.update.isPending && !nextProps.model.update.isPending;
-    let needsUpdate = !isEqual(this.props.initialValues, nextProps.initialValues);
+    let needsUpdate = !isEqual(this.props.model, nextProps.model);
     let { router } = this.context;
 
+    if ((nextProps.initialValues.isSelected !== this.props.initialValues.isSelected) ||
+    (nextProps.initialValues.isVisible !== this.props.initialValues.isVisible)) {
+      this.setState({
+        ...this.state,
+        packageSelected: nextProps.initialValues.isSelected,
+        packageVisible: nextProps.initialValues.isVisible
+      });
+    }
+
```

Comment:
curious about this piece here too, but again per our discussion, we will clarify with @elrickvm or @cowboyd later.